backend/api/simulation.py

```
@@ -2065,123 +2240,35 @@ def is_main_character_submit(persona_name, student_role):
+                        scene_goal=scene_goal,
+                        scene_description=scene_description,
+                        current_attempts=current_attempts,
+                        max_attempts=max_attempts,
+                        db=db,
+                        user_progress_id=user_progress.id,
+                        current_scene_id=scene_id_to_use,
+                        perform_db_progression=False
+                    )
+                    
+                    print(f"[DEBUG] Goal validation result: {validation_result}")
+                except Exception as e:
```

Ruff (0.14.2)

Do not catch blind exception: `Exception`

(BLE001)